### ISSUE TITLE
Filtering by secondary index between another index

### ISSUE THREAD

**Issue by JohnyDays** [rethinkdb/rethinkdb#4709]
# Scenario: 
- I have an event table, which is a log of events.
- This table has an index userId (string)
- This table has an index timestamp (date)
- I want to get all events for the user 4 from the last 2 days

# My expected solution:
```javascript
r.table('events')
  .between(r.now().sub(25600), r.now(), {index: 'timestamp'})
  .getAll('4', {index: 'residentId'})
```
or 
```javascript
r.table('events')
  .getAll('4', {index: 'residentId'})
  .between(r.now().sub(25600), r.now(), {index: 'timestamp'})
```

Neither work, is there an alternative way to do this? Am I missing something obvious?

**Comment by marshall007** [comment]
What you're trying to do is possible but the solution isn't that obvious [from the docs](http://rethinkdb.com/docs/secondary-indexes/javascript/#compound-indexes). You just need to create a compound secondary index on `[ residentId, timestamp ]` and use that index in your `.between()`. Here's your example rewritten this way:

```js
r.table('events').indexCreate('resident_timestamp', [ r.row('residentId'), r.row('timestamp') ])

r.table('events')
  .between(
    [ '4', r.now().sub(25600) ],
    [ '4', r.now() ],
    { index: 'resident_timestamp' }
  )
```

**Comment by danielmewes** [comment]
Another relevant issue is https://github.com/rethinkdb/rethinkdb/issues/809 , but that's a more long-term extension and it will probably never be as efficient as using a compound index.

**Comment by JohnyDays** [comment]
Awesome! I carefully read the docs, but I completely misunderstood the purpose of compound indexes. This solves the issue in a very elegant way. Do you think my use case is a common one? I might throw up a docs PR illustrating it on my free time if that's the case.

**Comment by danielmewes** [comment]
@JohnyDays a docs PR would be welcome! Is there a particular place in the docs where you think this should be explained?

**Comment by JohnyDays** [comment]
Probably as an example of a use for compound indexes? That is if you believe it is a common use case, filtering by one field and ordering by another,

**Comment by danielmewes** [comment]
Makes sense, I think that's not an uncommon use case at least.

**Comment by mholt** [comment]
Nice, I also ended up doing the exact same thing @marshall007 suggested above, and I also landed at the exact same query to get all the documents for a certain user within the recent period of time.

However, my query is returning all documents, even beyond that period of time. In other words, it's not honoring the `between()` for the `timestamp` field. 

When I query using `between()` on the single `timestamp` field (instead of the compound index with the user ID), it works fine.

Bug maybe? Or something subtle I'm missing?

```javascript
r.db("mydb").table("test")
  .between(
    [r.minval, r.now().sub(300)],  // all users, last 5 minutes
    [r.maxval, r.now()],
    {index: "user_timestamp"})
  .orderBy(r.desc("timestamp"))
  .limit(10)
```

^^ Should only return 2 rows, but returns 10.
